lib/fabrik/DirectorManager.js

```diff
@@ -212,7 +213,11 @@ class DirectorManager extends BaseManager {
           return;
         }
       })
-      .then(() => this.generateManifest(deploymentName, opts))
+      .then(() => action === CONST.OPERATION_TYPE.CREATE ?
+        this.executeActions(CONST.SERVICE_LIFE_CYCLE.PRE_CREATE, deploymentName, params, args) :
+        this.executeActions(CONST.SERVICE_LIFE_CYCLE.PRE_UPDATE, deploymentName, params, args)
+      )
```

Comment:
to keep the above concise and avoid repetition of same params, can be done sth like the below:
```
let serviceLifeCycle; //Defined in line #185

servcieLifeCycle = CONST.SERVICE_LIFE_CYCLE.PRE_CREATE; //set in line #202
servcieLifeCycle = CONST.SERVICE_LIFE_CYCLE. PRE_UPDATE; //set in line #190

.then(() =>  this.executeActions(serviceLifeCycle, deploymentName, params, args));// This line replaces the existing tertiary check at #216


```

Reply:
Done!